modules/seqexec/engine/src/main/scala/seqexec/engine/UserEvent.scala

```diff
@@ -33,7 +33,7 @@ final case class GetState[D<:Engine.Types](f: D#StateType => Option[Stream[IO, E
 }
 // Generic event to put a function in the main Process process, which changes the state
 // depending on the current state
-final case class ModifyState[D<:Engine.Types](f: D#StateType => (D#StateType, D#EventData)) extends UserEvent[D] {
+final case class ModifyState[D<:Engine.Types](f: Engine.HandleP[D#StateType, Event[D], D#EventData]) extends UserEvent[D] {
```

Comment:
Changing `f` to a `HandleP` means that `f` now can produce a feedback `Stream`. This allows, for example, to start a sequence inside `f`.